borg/helpers.py

```diff
@@ -1160,3 +1162,49 @@ def write(self, s):
                 except OSError:
                     pass
         return len(s)
+
+
+class SignalException(BaseException):
+    """base class for all signal-based exceptions"""
+
+
+class SigHup(SignalException):
+    """raised on SIGHUP signal"""
+
+
+class SigTerm(SignalException):
+    """raised on SIGTERM signal"""
+
+
+@contextlib.contextmanager
+def signal_handler(sig, handler):
```

Comment:
this is based on code already present in master, but was functionally enhanced to support giving signal NAMES and not failing when a signal name is not supported on some platform.